### PR TITLE
[ALLUXIO-1724] Adding mount point field to FileInfo

### PR DESCRIPTION
**Pull request by jsimsa**
Addresses: https://tachyon.atlassian.net/browse/ALLUXIO-1724

### CODE REVIEW

**Comment by jsimsa** [comment]
@gpang added the journal field ... are there are test resources / classes I need to modify?

**Comment by gpang** [comment]
I think you already changed the test resource I am aware of.

**Comment by gpang** [comment]
LGTM

File: core/server/src/main/java/alluxio/master/file/meta/InodeFile.java [hunk 1]
```diff
@@ -326,14 +327,15 @@ public synchronized void complete(long length)
...
-    sb.append(super.toString()).append(", LENGTH: ").append(mLength);
-    sb.append(", Cacheable: ").append(mCacheable);
-    sb.append(", Completed: ").append(mCompleted);
-    sb.append(", Cacheable: ").append(mCacheable);
-    sb.append(", mBlocks: ").append(mBlocks);
-    sb.append(", mTtl: ").append(mTtl);
-    sb.append(")");
-    return sb.toString();
+    return sb.append(super.toString())
+        .append(", BLOCKS: ").append(mBlocks)
```
Comment (yupeng9) [inline core/server/src/main/java/alluxio/master/file/meta/InodeFile.java]: Is it the new convention to make all the fields uppercase?
I feel it's easier just to use Guava's toStringHelper: http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/base/Objects.ToStringHelper.html

Comment (jsimsa) [inline core/server/src/main/java/alluxio/master/file/meta/InodeFile.java]: This was for consistency with the super.toString() output, but I agree with you. I will make the change to toStringHelper.

File: core/server/src/main/java/alluxio/master/file/meta/options/CreatePathOptions.java [hunk 2]
```diff
@@ -90,6 +92,16 @@ public Builder setDirectory(boolean directory) {
     }
 
     /**
+     * @param mountPoint the mount point flag to use; it specifies whether the object to create is
```
Comment (yupeng9) [inline core/server/src/main/java/alluxio/master/file/meta/options/CreatePathOptions.java]: Why is this needed? Wouldn't be mount command the one that specifies this info?

File: core/server/src/proto/journal/file.proto [hunk 3]
```diff
@@ -35,7 +35,7 @@ message DeleteMountPointEntry {
   optional string alluxio_path = 1;
 }
 
-// next available id: 11
+// next available id: 12
```
Comment (yupeng9) [inline core/server/src/proto/journal/file.proto]: Do  we need to consider the versioning?

Comment (jsimsa) [inline core/server/src/proto/journal/file.proto]: this is a backwards compatible change

File: core/server/src/main/java/alluxio/master/file/meta/options/CreatePathOptions.java (hunk 2 above)
Comment (jsimsa) [inline core/server/src/main/java/alluxio/master/file/meta/options/CreatePathOptions.java]: not sure I understand your comment ... the mount function ends up invoking loadMetadataDirectory, which ends up invoking mkdir, which ends up invoking createPath which uses these options ... we need to propagate the "mount point" bit from loadMetadataDirectory to createPath.

Comment (yupeng9) [inline core/server/src/main/java/alluxio/master/file/meta/options/CreatePathOptions.java]: gotcha. Thanks for the explanation.

File: core/server/src/main/java/alluxio/master/file/meta/InodeFile.java (hunk 1 above)
Comment (jsimsa) [inline core/server/src/main/java/alluxio/master/file/meta/InodeFile.java]: done

**Comment by yupeng9** [comment]
LGTM. btw, given the size of this PR, I don't think it's a smallfix. So you may want to create a ticket to track it.

**Comment by jsimsa** [comment]
@yupeng9 thanks, created a ticket ... it started as a small fix :-)

**Comment by yupeng9** [comment]
lol, I can imagine it :)
